### PR TITLE
Fix/1833 slider input

### PR DESCRIPTION
**Pull request by kwongz**
### Checklist

image

A few notes, there is a Tick functionality that i think will make for a better user experience. Would like to try tackling to add that feature at a later date.

Would also like to try adding a live sql graph or table to connect the slider to.

- [ x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [ x] I have added to the docs where applicable
- [ ] I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable

### CODE REVIEW

File: sites/docs/pages/components/slider.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+````markdown
+<Slider
+    title="Months" 
+    name=monthsLarge
+    min=0
+    max=36
+    size=large
+/>
+````
+
```
Comment (archiewood) [inline sites/docs/pages/components/slider.md:98]: Would be great to add a proplisting table similar to https://docs.evidence.dev/components/dropdown/

File: packages/ui/core-components/src/lib/atoms/inputs/slider/Slider.svelte [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		return sizeMap[size.toLowerCase()] || 'w-40';
+	};
+
+	$: sizeClass = renderSize(size);
+</script>
+
+<HiddenInPrint enabled={hideDuringPrint}>
+	<div class={`relative ${sizeClass} mb-8`}>
+		<p class="pb-2 truncate text-xs">
+			{title} : <span class="text-sm">{$inputs[name]}</span>
```
Comment (archiewood) [inline packages/ui/core-components/src/lib/atoms/inputs/slider/Slider.svelte:92]: perhaps the title and the number should be the same font size?

File: packages/ui/core-components/src/lib/atoms/inputs/slider/Slider.svelte [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		</p>
+		<SliderShadcn {min} {max} {step} {ticks} bind:value />
```
Comment (archiewood) [inline packages/ui/core-components/src/lib/atoms/inputs/slider/Slider.svelte:94]: The slider position does not appear to currently be reactive to changes in the slider value

CleanShot 2024-05-28 at 15 29 47

I think the linking of two sliders to the same input `monthsLarge` in the docs was probably unintentional, but it does illustrate a reactivity quirk

**Review by archiewood — COMMENTED** [review]
Loving the slider functionality. 

A few comments inline on UX

File: sites/docs/pages/components/slider.md (hunk 1 above)
Comment (archiewood) [reply sites/docs/pages/components/slider.md:98]: (the checkbox is definitely a bit lean on docs content)

File: packages/ui/core-components/src/lib/atoms/inputs/slider/Slider.svelte (hunk 3 above)
Comment (kwongz) [reply packages/ui/core-components/src/lib/atoms/inputs/slider/Slider.svelte:94]: interesting, resolving the linkage and taking a look at the reactivity quirk

File: packages/ui/core-components/src/lib/atoms/inputs/slider/Slider.svelte (hunk 2 above)
Comment (kwongz) [reply packages/ui/core-components/src/lib/atoms/inputs/slider/Slider.svelte:92]: Will do!

File: sites/docs/pages/components/slider.md (hunk 1 above)
Comment (kwongz) [reply sites/docs/pages/components/slider.md:98]: ill take a look at the dropdown table thanks!

Comment (kwongz) [reply sites/docs/pages/components/slider.md:101]: Props Table Added!

File: packages/ui/core-components/src/lib/atoms/inputs/slider/Slider.svelte (hunk 3 above)
Comment (kwongz) [reply packages/ui/core-components/src/lib/atoms/inputs/slider/Slider.svelte]: fixed the Large and Full slider connection

Was not able to experience the reactivity quirk on my side, would like to talk more about it when we can.

Thanks!

Comment (archiewood) [reply packages/ui/core-components/src/lib/atoms/inputs/slider/Slider.svelte]: I think you will need to keep the erroneous linkage in order to reproduce the issue.

this was on Firefox, for what it's worth

**Review by mcrascal — APPROVED** [review]
This is looking great. 

A couple ideas: 

1. Can we apply the class `select-none` so that when you're dragging around the slider, it doesn't select it like text
2. Can we add support for the `fmt` prop on this (e.g. a percent slider, a $ slider etc)

**Comment by kwongz** [comment]
Updated Slider

https://www.loom.com/share/a3c1eed75187401a87856303b8e8fce5?sid=ed1f5e87-bb49-46c1-8a5a-407e9a79adef

**Pull request merged**